liquid-rust-typeck/src/checker.rs

```diff
@@ -471,6 +471,28 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
 
     fn check_bitwise_op(&self, op: BinOp, ty1: Ty, ty2: Ty) -> Ty {
         match (ty1.kind(), ty2.kind()) {
+            (
+                TyKind::Refine(BaseTy::Int(int_ty1), _e1),
+                TyKind::Refine(BaseTy::Int(int_ty2), _e2),
+            ) => {
+                debug_assert_eq!(int_ty1, int_ty2);
+                TyKind::Refine(
+                    BaseTy::Int(*int_ty1),
+                    ExprKind::Constant(liquid_rust_fixpoint::Constant::Bool(true)).intern(),
+                )
+                .intern()
```

Comment:
This is not well-formed, it should be:

```suggestion
TyKind::Exists(BaseTy::Int(*int_ty1), Expr::tt().into()).intern()
```

Reply:
Ah, that looks much better, thanks.